src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -8205,6 +8205,312 @@ void MacroAssembler::evmovdqu(BasicType type, KRegister kmask, Address dst, XMMR
   }
 }
 
+void MacroAssembler::kand(BasicType type, KRegister dst, KRegister src1, KRegister src2) {
+  switch(type) {
+    case T_BOOLEAN:
+    case T_BYTE:
+       kandbl(dst, src1, src2);
```

Comment:
Is the type here the vector element type? Is so, for byte vector we need kandql. For short kanddl. For int kandwl, For long kandbl.

Reply:
No its not vector type. From the instruction side always T_LONG basic type is passed. TypeVectMask is vector of booleans.  We can use vector length to emit shorter instructions ,  but efficiency wise its wont be different to doing an operation over entire opmask register.